dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/dataitems/DataItemQueryTests.java

```diff
@@ -0,0 +1,297 @@
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsWithoutPagination()
+    {
+        // Given
+        final String noPagination = "?paging=false";
+
+        // When
+        final ApiResponse response = dataItemActions.get( noPagination );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", is( nullValue() ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPaginationOrderedByCode()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get( "?order=code:asc" );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", isA( String.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
+    }
+
+    @Test
+    public void testFilterByDimensionTypeUsingDefaultPagination()
+    {
+        // Given
+        final String theDimensionType = "PROGRAM_INDICATOR";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, theDimensionType ) );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", is( theDimensionType ) );
+    }
+
+    @Test
+    public void testFilterByProgramUsingDefaultPagination()
+    {
+        // Given
+        final String theDimensionType = "PROGRAM_INDICATOR";
+        final String theProgramId = "BJ42SUrAvHo";
+        final String aValidFilteringAttribute = "program.id";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]&filter=" + aValidFilteringAttribute
+            + ":eq:%s&order=code:asc";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, theDimensionType, theProgramId ) );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
```

Comment:
Are you sure that this one will always be first item returned? You could do `"nameableObjects.code", contains('...')`

Reply:
Done